### PR TITLE
Added missing steps of cloning "paho" repo.

### PR DESCRIPTION
**Pull request by paTelshREya**
Branch: shreya/improving-docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,14 +43,22 @@ Then compile the MQTT-SN enabled CLI publishing example as follows
 $ git clone https://github.com/openthread/ot-nrf528xx.git
 $ cd ot-nrf528xx && rm -Rf openthread
 $ git clone https://github.com/DynamicDevices/openthread.git -b ajl/adding-examples
```
Comment (ajlennon) [inline README.md]: @paTelshREya I don't think we should need to clone paho below. Instead we should add --recursive to the end of this line. Can you try?

Comment (paTelshREya) [reply README.md]: Yes. That worked.

Comment (ajlennon) [reply README.md]: ok can you commit that into your branch and push and it should automatically update this PR :)

**Review by ajlennon — APPROVED** [review]
Looks good

**Pull request merged**